Assets/CharTest/Scripts/SeleneFire.cs

```diff
@@ -0,0 +1,117 @@
+
+    private Vector2 startShot;
+    [HideInInspector]
+    public Vector2 TarDir;
+    public CharacterTargeting CharTar;
+
+    [HideInInspector]
+    public List<IBaseBullet> bulletList;
+
+    private SharedCharacterController charCon;
+     // Start is called before the first frame update
+    void Start()
+    {
+        CharTar = GetComponent<CharacterTargeting>();
+        bulletList = new List<IBaseBullet>();
+        charCon = GetComponent<SharedCharacterController>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {   
+        startShot = transform.position;
+        TarDir = CharTar.TargetDirection();
+ 
+         if (SpellMap.Instance.GetSpellDown(charCon.playerNumber, SpellType.PRIM))       
+        {
+            StartCoroutine(SelenePrimary(primaryProjectiles, primaryWaves, startShot));
+        }
+     
+        if (SpellMap.Instance.GetSpellDown(charCon.playerNumber, SpellType.HEAVY))
+        {  
+            StartCoroutine(SeleneHeavy(heavyProjectiles));
+        }
+
+        if (SpellMap.Instance.GetSpellDown(charCon.playerNumber, SpellType.INTRIN))
+        {
+            for (int i = 0; i < bulletList.Count; i++)
+            {
+                bool mutateCheck = bulletList[i].CheckMutation();
+                if (mutateCheck != true)
+                {
+                    StartCoroutine(bulletList[i].IntrinsicMutate());
+                }
+            }   
+        }
+    }   
+
+    IEnumerator SelenePrimary(int _primaryProjectiles, int _primaryWaves, Vector2 startShot)
+    {
+        int tempProjectiles = primaryProjectiles;
+        float baseAngle = Mathf.Atan2(TarDir.y, TarDir.x) * Mathf.Rad2Deg;
+
+        for (int i = primaryWaves; i > 0; i--)  
+       {
+            float angle = baseAngle - (60f/2);
+            for (int k = 0; k < tempProjectiles; k++)
+            {
+                float shotDirXPos = Mathf.Cos(angle * Mathf.Deg2Rad);
+                float shotDirYPos = Mathf.Sin(angle * Mathf.Deg2Rad);
+
+                Vector2 shotDirection = new Vector2(shotDirXPos, shotDirYPos) * primarySpeed;
+
+                GameObject tempObj = Instantiate(primaryPrefab, startShot, Quaternion.identity);
+                PrimaryBullet primaryBullet = tempObj.GetComponent<PrimaryBullet>();
+                bulletList.Add(primaryBullet);
+                primaryBullet.player = this;
+                primaryBullet.shotOrigin = startShot;
+                tempObj.GetComponent<Rigidbody2D>().velocity = shotDirection;
+                angle += 60f / (tempProjectiles + 1);
+
+            }
+            yield return new WaitForSeconds(.4f);
+            tempProjectiles = tempProjectiles - 1; 
+        }
+        
+    }
+
+    IEnumerator SeleneHeavy(int _heavyProjectiles)
+    {
+        float angle = Mathf.Atan2(TarDir.y, TarDir.x) * Mathf.Rad2Deg;
+        float shotDirXPos = Mathf.Cos(angle * Mathf.Deg2Rad);
+        float shotDirYPos = Mathf.Sin(angle * Mathf.Deg2Rad);
```

Comment:
This Vector -> Angle -> Vector code is unnecessary as far as I can tell. Just use TarDir as the direction vector.